src/resolver/descriptor.cc

```diff
@@ -189,6 +190,189 @@ initManifestDescriptorAbsPath( ManifestDescriptor &          desc,
     }
 }
 
+/* -------------------------------------------------------------------------- */
+
+void
+from_json( const nlohmann::json &jfrom, ManifestDescriptorRaw &descriptor )
+{
+  if ( ! jfrom.is_object() )
+    {
+      std::string aOrAn = jfrom.is_array() ? " an " : " a ";
+      throw ParseManifestDescriptorRawException(
+        "manifest descriptor must be an object, but is" + aOrAn
+        + std::string( jfrom.type_name() ) + '.' );
+    }
+
+  /* Clear fields. */
+  descriptor.name = std::nullopt;
```

Comment:
non-blocking TODO: we should add `void ManifestDescriptorRaw::clear();` in the future, but this is perfectly fine.

Reply:
Added todo